test/test_restful.py

```diff
@@ -915,6 +946,49 @@ def test_060_cv_version_get(self):
         self.assertEqual(results['current_version'], api_version.current_version())
         self.assertEqual(results['supported_versions'], api_version.all_versions())
 
+    # Rust agent testset
+    @unittest.skipIf(RUST_TEST, "Testing against rust-keylime is disabled!")
+    def test_070_rust_agent_setup(self):
+        """Set up the Rust agent"""
+
+        # Kill the Python agent and launch the Rust agent!
+        kill_cloudagent()
+        launch_rust_cloudagent()
```

Comment:
you launch the agent as a side effect of this test, you can consider to put it in `setUp`, and the kill in `tearDown`

Reply:
Putting the invocation in `setUp` and `tearDown` means we'd have to restart the agent every single test run, which would be extremely slow. Plus, some tests require the agent to be running between them, so that wouldn't work anyway.

It _might_ make sense to do it in `setUpClass`, but since we have to run separate versions of the agent at different times for different tests, that doesn't work at the moment either.

It's in a separate test so we know it's invoked before the rest of the Rust test suite. This is necessary because one of the reused tests (020) includes its own call to `launch_cloudagent`. If the Rust agent isn't running when that call happens, the Python agent will be booted up instead.

Reply:
Even if we can't put it in `setUp()` and `tearDown()` it might make sense to pull it out into it's own function that checks to see if the agent is running and if not, starts the agent (if it's not running, and does a loop to make sure it's up before returning). Then this function can be included in every test that needs that agent. This would allow developers to be able to run single test methods as they are developing (a common pattern in most projects).

Then we could handle stopping the agents in the class teardown if they are running.

I know this is a change to the way things were for the other agent (the existing tests were fragile in this way), but it's becoming more necessary to fix if we copy that fragility into more places.

Reply:
@mpeters Okay, I'll give that a try. I think the cleanest way might actually just be to improve the existing `launch_cloudagent` function - it already checks if there's an agent running. If we provide a parameter to toggle between launching the Python and Rust agents, and incorporate @aplanas' suggestion around waiting until it's up, we could just include it in front of all tests that need an agent, as you say. Then, for the Rust tests, we just have one extra that stops it and restarts the Rust agent - that should default to Python for now and allow Rust when needed.